dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java

```diff
@@ -1132,6 +1162,151 @@ public void shouldReturnParserContainerUUID() throws Exception {
+                        .setUser(systemUser)
+                        .setPageUri(pageEnglishVersion.getURI())
+                        .setPageMode(PageMode.LIVE)
+                        .build(),
+                mockRequest, mockResponse);
+
+        Assert.assertEquals("content2content1", html);
+    }
+
+    /**
+     * Method to test: {@link com.dotmarketing.portlets.htmlpageasset.business.render.HTMLPageAssetRenderedAPIImpl#getPageHtml(PageContext, HttpServletRequest, HttpServletResponse)}
+     * Given Scenario: Create a page and a not advance template using a File Container in another site
+     * ExpectedResult: should work
+     *
+     * @throws Exception
+     */
+    @Test
+    public void shouldRenderUsingOtherSiteContainerAndNotAdvanceTemplate() throws Exception {
+        final Host defaultHost = APILocator.getHostAPI().findDefaultHost(APILocator.systemUser(), true);
+        final FileAssetContainer container = createFileContainer(defaultHost);
+
+        final TemplateLayout templateLayout = new TemplateLayoutDataGen()
+                .withContainer("//" + defaultHost.getName()  + container.getPath())
+                .next();
+
+        final Contentlet contentlet = new ThemeDataGen().nextPersisted();
+        Template template = new TemplateDataGen()
+                .title("PageContextBuilderTemplate"+System.currentTimeMillis())
+                .host(site)
+                .drawedBody(templateLayout)
+                .theme(contentlet)
+                .nextPersisted();
+
+        PublishFactory.publishAsset(template, systemUser, false, false);
+
+        final String pageName = "testPage-"+System.currentTimeMillis();
+        final HTMLPageAsset pageEnglishVersion = createHtmlPageAsset(template, pageName, 1);
+
+        createMultiTree(pageEnglishVersion.getIdentifier(), container.getIdentifier(), "1");
+
+        final HttpServletRequest mockRequest = createHttpServletRequest(pageEnglishVersion);
+        Mockito.when(mockRequest.getParameter(WebKeys.PAGE_MODE_PARAMETER)).thenReturn(PageMode.LIVE.toString());
+
+        final HttpServletResponse mockResponse = mock(HttpServletResponse.class);
+        final HttpSession session = createHttpSession(mockRequest);
+        Mockito.when(session.getAttribute(WebKeys.VISITOR)).thenReturn(null);
+        Mockito.when(session.getAttribute(WebKeys.CMS_USER)).thenReturn(systemUser);
+
+        final String html = APILocator.getHTMLPageAssetRenderedAPI().getPageHtml(
+                PageContextBuilder.builder()
+                        .setUser(systemUser)
+                        .setPageUri(pageEnglishVersion.getURI())
+                        .setPageMode(PageMode.LIVE)
+                        .build(),
+                mockRequest, mockResponse);
+
+        Assert.assertTrue(html.contains("content2content1"));
+    }
+
+    @NotNull
+    private HttpServletRequest createHttpServletRequest(HTMLPageAsset pageEnglishVersion) throws DotDataException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'pageEnglishVersion' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5067852)